src/conn/conn_cache_pool.c

```diff
@@ -453,22 +468,41 @@ __cache_pool_assess(WT_SESSION_IMPL *session, uint64_t *phighest)
 			continue;
 		cache = entry->cache;
 		++entries;
-		new = cache->bytes_read;
+		/* Copy the value out so it doesn't change underneath us */
+		tmp = cache->bytes_read;
 		/* Handle wrapping of eviction requests. */
-		if (new >= cache->cp_saved_read)
-			cache->cp_current_read = new - cache->cp_saved_read;
+		if (tmp >= cache->cp_saved_read)
+			cache->cp_pass_read = tmp - cache->cp_saved_read;
 		else
-			cache->cp_current_read = new;
-		cache->cp_saved_read = new;
-		if (cache->cp_current_read > highest)
-			highest = cache->cp_current_read;
+			cache->cp_pass_read = tmp;
+		cache->cp_saved_read = tmp;
+
+		/* Update the application eviction count information */
+		tmp = cache->cp_saved_app_evicts;
+		cache->cp_pass_app_evicts =
+		    tmp - cache->cp_pass_app_evicts;
```

Comment:
I don't follow this calculation.  Let's say there are 10 evictions happening between each pass.  Then we would have:

pass 1: saved = 0, pass = 10 --> saved = -10
pass 2: saved = -10, pass = 10 --> saved = -20

Am I misunderstanding the logic?

etc.